chall06/vsempere.c

```diff
@@ -0,0 +1,218 @@
+}
+
+static int      ft_check_south_west(char *board, int ring, int board_size, int king)
+{
+    int         row_chk;
+    int         max;
+
+    max = board_size * board_size;
+    row_chk = king + (ring *  board_size);
+    
+    return (((row_chk) % board_size > (row_chk - ring) % board_size) && 
+        (row_chk - ring) >= 0 &&
+        (board[row_chk - ring] & (VSEMPERE_B | VSEMPERE_Q | (ring == 1 ? VSEMPERE_P : 0))) ? 1 : 0);
+}
+
+static int      ft_check_north_east(char *board, int ring, int board_size, int king)
+{
+    int         row_chk;
+    int         max;
+
+    max = board_size * board_size;
+    row_chk = king - (ring *  board_size);
+    
+    return (((row_chk) % board_size < (row_chk + ring) % board_size) && 
+        (row_chk + ring) < max &&
+        (board[row_chk + ring] & (VSEMPERE_B | VSEMPERE_Q)) ? 1 : 0);
+}
+
+static int      ft_check_north(char *board, int ring, int board_size, int king)
+{
+    int         row_chk;
+    int         max;
+
+    max = board_size * board_size;
+    row_chk = king - (ring *  board_size);
+    
+    return (row_chk >= 0 &&
+        (board[row_chk] & (VSEMPERE_Q | VSEMPERE_R)) ? 1 : 0);
+}
+
+static int      ft_check_south(char *board, int ring, int board_size, int king)
+{
+    int         row_chk;
+    int         max;
+
+    max = board_size * board_size;
+    row_chk = king + (ring *  board_size);
+    
+    return (row_chk < max &&
+        (board[row_chk] & (VSEMPERE_Q | VSEMPERE_R)) ? 1 : 0);
+}
+
+static int      ft_check_east(char *board, int ring, int board_size, int king)
+{
+    int         row_chk;
+    int         max;
+
+    max = board_size * board_size;
+    row_chk = king + ring;
+    
+    return (((king) % board_size < (row_chk) % board_size) && 
+        row_chk < max &&
+        (board[row_chk] & (VSEMPERE_Q | VSEMPERE_R)) ? 1 : 0);
+}
+
+static int      ft_check_west(char *board, int ring, int board_size, int king)
+{
+    int         row_chk;
+    int         max;
+
+    max = board_size * board_size;
+    row_chk = king - ring;
+    
+    return (((king) % board_size > (row_chk) % board_size) && 
+        row_chk >= 0 &&
+        (board[row_chk] & (VSEMPERE_Q | VSEMPERE_R)) ? 1 : 0);
+}
+
+static int     ft_check_mate_ring(char *board, int king,
+    int ring, int board_size)
+{
+    if (ring >= board_size)
+        return (2);
+
+    return (ft_check_south_east(board, ring, board_size, king) || 
+        ft_check_north_west(board, ring, board_size, king) ||
+        ft_check_south_west(board, ring, board_size, king) ||
+        ft_check_north_west(board, ring, board_size, king) ||
+        ft_check_north_east(board, ring, board_size, king) ||
+        ft_check_north(board, ring, board_size, king) ||
+        ft_check_south(board, ring, board_size, king) ||
+        ft_check_east(board, ring, board_size, king) ||
+        ft_check_west(board, ring, board_size, king) ? 1 : 0);
+}
+
+static int      ft_check_map(char *board)
```

Comment:
There's some missing checks in this function, like for example, a `\n` character at the end or the beginning of the board would make it invalid, or also this `board = "K..\n\n...\nR.."` is invalid cause of double `\n`. But this tests are a bit hardcore so I'm not sure they will be added.